disco_aws_automation/disco_deploy.py

```diff
@@ -390,6 +391,103 @@ def handle_tested_ami(self, old_hostclass_dict, ami, desired_size,
+
+        # Spinup our new autoscaling group in testing mode, making one even if one already exists.
+        self._disco_aws.spinup([new_group_config], create_if_exists=True, testing=True)
+
+        new_group = self._disco_autoscale.get_existing_group(hostclass=hostclass, throw_on_two_groups=False)
+
+        if old_group and old_group.name == new_group.name:
+            raise RuntimeError("Old group and new group should not be the same.")
+
+        try:
+            if (self.wait_for_smoketests(ami.id, new_group_config["desired_size"] or 1) and
+                    (not run_tests or self.run_integration_tests(ami, wait_for_elb=uses_elb))):
+                # If testing passed, mark AMI as tested
+                self._promote_ami(ami, "tested")
+                # Get list of instances in group
+                group_instance_ids = [inst.instance_id for inst in
+                                      self._disco_autoscale.get_instances(group_name=new_group.name)]
+                group_instances = self._disco_aws.instances(instance_ids=group_instance_ids)
+                # If we are actually deploying and are able to leave testing mode and start serving requests
+                if deployable and self._set_testing_mode(hostclass, group_instances, False):
+                    logging.info("Successfully left testing mode for group %s", new_group.name)
+                    # Update ASG to exit testing mode and attach to the normal ELB if applicable.
+                    self._disco_aws.spinup([new_group_config], group_name=new_group.name)
+                    if uses_elb:
+                        try:
+                            # Wait until the new ASG is registered and marked as healthy by ELB.
+                            self._disco_elb.wait_for_instance_health_state(hostclass=hostclass,
+                                                                           instance_ids=group_instance_ids)
+                        except TimeoutError:
+                            logging.exception("Waiting for health of instances attached to ELB timed out")
+                            # Destroy the testing ASG
+                            self._disco_autoscale.delete_groups(group_name=new_group.name, force=True)
+                            if uses_elb:
+                                # Destroy the testing ELB
+                                self._disco_elb.delete_elb(hostclass, testing=True)
+                            return
+                    # we can destroy the old group
+                    if old_group:
+                        # Destroy the original ASG
+                        self._disco_autoscale.delete_groups(group_name=old_group.name, force=True)
+                else:
+                    # Otherwise, we need to keep the old group and destroy the new one
+                    if deployable:
+                        reason = "Unable to exit testing mode for group {}".format(new_group.name)
```

Comment:
Something like not being able to change the 'is_testing' tag or not being able to rename the nerve groups to come up as the expected nerve service.